routes/items.js

```diff
@@ -18,14 +22,17 @@ router.post("/", (req, res, next) => {
 });
 
 router.put("/:id", (req, res, next) => {
-  // TODO: complete writing this route handler
-  res.status(501).send({ error: 'route not yet implemented' });
+  res.status(200).send(itemData.updateById(req.params.id, req.body));
+  
 });
 
-
 router.delete("/:id", (req, res, next) => {
-  // TODO: complete writing this route handler
-  res.status(501).send({ error: 'route not yet implemented' });
+ if(itemData.getById(req.params.id)){
+  res.status(200).send(itemData.deleteById(req.params.id));
+ }
+  else {
+    res.status(404).json({error: "item does not exist"})
+  }
```

Comment:
The logic of whether the item exists should be handled by the data interface method. The logic of what to do based on whether the item is found is great. Maybe `deleteById` could return either the deleted item or null, depending on whether it's found?

Reply:
Cheri, I am bit confused here how it makes difference if we handle item found condition in the data interface method or in the router method handler ? is it about best practices ?